build.gradle

```diff
@@ -52,6 +60,14 @@ tasks.withType(JavaCompile) {
     options.compilerArgs += ['-Werror', '-Xlint:deprecation']
 }
 
+// Access groovy from java sourceset
+sourceSets {
+    main {
+        java { srcDirs = [] }
+        groovy { srcDirs += ['src/main/java'] }
+    }
+}
```

Comment:
ooh hacky :P

Reply:
I thought this might catch your eye, but I didn't really want to move all the existing code into `groovy`, at least not in this PR.

Reply:
It just occurred to me that doing things this way means we don't run errorprone on anything (and we should probably fix it in other places where we do this).
Can we put this groovy file in a different source set perhaps that we expose through main?


```groovy
sourceSets {
  groovy
  main.output.dir groovy.output
}

// one/both of these might not be necessary
configurations.runtimeOnly.extendsFrom configurations.groovyRuntimeClasspath
configuration.implementation.extendsFrom configurations.groovyCompile
```